.github/workflows/brs-commit-checker.yml

```diff
@@ -0,0 +1,93 @@
+name: BRS Commit Checker Report
+on:
+      workflow_dispatch:
+        inputs:
+            fix_version:
+                type: string
+                required: true
+                description: The ICU Jira "Fix Version" semver
+            from_git_ref:
+                type: string
+                required: true
+                description: The git ref start of comparison range. Prefix branches with `origin/`.
+            end_git_ref:
+                type: string
+                required: true
+                description: The git ref end of comparison range. Must be descendant of `from_git_ref`. Prefix branches with `origin/`.
+            # Jira user name & API token is used for processing sensitive tickets
+            jira_user_email:
+                type: string
+                required: false
+                description: (optional to include sensitive tickets) Email address of Jira user
+            # Jira user name & API token is used for processing sensitive tickets
+            # We protect the API token as a secret by preventing its value from pasted at the command line
+            # by using add-mask to obfuscate secret values in the jobs
+            # https://docs.github.com/en/actions/writing-workflows/choosing-what-your-workflow-does/workflow-commands-for-github-actions
+            jira_user_api_token:
+                type: string
+                required: false
+                description: (optional to include sensitive tickets) API token of Jira user
+
+jobs:
+    commit-report:
+        runs-on: ubuntu-latest
+        steps:
+            - uses: actions/checkout@v4
+              with:
+                fetch-tags: true
+                fetch-depth: 0
+            # workaround for bug in checkout action. this step should be redundant.
+            # https://github.com/actions/checkout/issues/1471
+            # https://github.com/actions/checkout/issues/1781
+            # https://github.com/actions/checkout/issues/701#issuecomment-1133937950
+            - name: Fetch all tags
+              run: |
+                git fetch --tags origin
+            - name: Fetch all branches
+              run: |
+                git fetch origin
+            - name: Setup Python
+              uses: actions/setup-python@v5
+              with:
+                python-version: '3.12.8'
+                cache: 'pipenv'
+                cache-dependency-path: |
+                    tools/commit-checker/Pipfile
+                    tools/commit-checker/Pipfile.lock
+            - name: Install pipenv
+              run: |
+                sudo pip3 install pipenv
```

Comment:
Can't `venv` do the same kind of things (out of the box, no need to install anything)

I did a very quick scan of several articles and there seems to be agreement that `pipenv` is more powerful.

But I've also seen several recommendations to start with `venv`, get familiar with virtual environments, and when/if you need more power switch to `pipenv`.

More power often means added complexity :-)

Reply:
I considered that out of scope. This is just automating the existing instructions. 

I also don't know enough about either to comment, but I would be interested to learn more, assuming that this topic is orthogonal to the objective of the PR.

Reply:
I wrote this script and chose pipenv due to its simplicity and reproducibility. It behaves more like a modern package manager. I've tried many ways to install Python dependencies and this is by far my favorite.

Reply:
> I don't like venv because you make a profile global to the system user

I am not sure I understand this.

With venv I create a virtual environment in whatever folder I want.
I can even create it under my file tree covered by a git repo.
There is nothing global, or system user.

> behaves more like a modern package manager

Not necessarily a plus in my book :-)

---

Not enough to block the pr on this.
It's in a cloud machine, whatever.